src/main/java/com/kytheralabs/SeleniumScripter.java

```diff
@@ -338,6 +339,29 @@ private void runWait(Map<String, Object> script) throws Exception {
             }
 
             System.out.println("Object found");
+        if (script.containsKey("asyncwait") && script.get("asyncwait").equals(true)) {
+            //To set the script timeout to 10 seconds
+            driver.manage().timeouts().setScriptTimeout(10, TimeUnit.SECONDS);
+            //To declare and set the start time
+            long startTime = System.currentTimeMillis();
+            //Calling executeAsyncScript() method to wait for 10 seconds
+            js.executeAsyncScript("window.setTimeout(arguments[arguments.length - 1], 10000);");
+            //To get the difference current time and start time
+            System.out.println("Wait time: " + (System.currentTimeMillis() - startTime));
+        }else if (script.containsKey("refresh") && script.get("refresh").equals(true)) {
+            System.out.println("Refreshing page");
+            //Calling executeAsyncScript() method to refresh page
+            js.executeScript("history.go(0)");
+            //waits for page to load
+            js.executeAsyncScript("window.setTimeout(arguments[arguments.length - 1], 10000);");
+            System.out.println("Page refreshed");
+        }else if (script.containsKey("back") && script.get("back").equals(true)) {
+            System.out.println("Going to last page");
+            //Calling executeAsyncScript() method to go back a page
+            js.executeScript("window.history.back();");
+            //waits for page to load
+            js.executeAsyncScript("window.setTimeout(arguments[arguments.length - 1], 10000);");
+            System.out.println("Page refreshed");
```

Comment:
To my untrained eye, this all seems to be in the runWait block, how does it work from a YAML standpoint? because that doesn't seem correct

Reply:
I basically got the idea from how you created the "failnotfound": false.
You'd basically just add 
"asyncwait": true to a normal time block or "back" or "refresh" in the same way to get it to work.

Reply:
yeah but back and refresh isn't waiting, is it?

Reply:
Very great point. I'll remove those instances and just leave the wait. I'll see where I can apply those later when I need them but for now I'll get rid of the extra stuff.

Reply:
Cool, I mean they all look good and I'm happy to have them in the platform and the vids look great, but if they go in here, they'll never leave and it seems like the wrong place to put back and refresh, thats all.

Reply:
Totally agree.